### PR TITLE
7 all activities for one trip

### LINKED ISSUE #7: All Activities for one Trip
Get /api/v1/itinerary/:id/activities If valid itinerary id, return accommodation for that trip and all the details for that activity

```
status: 200
{
  "data": [
    {
      "id": "1",
      "type": "activities"
      "attributes": {
        "name": "Sushi Palace"
        "address": "123 Main Street, Phuket, Thailand",
        "description": "Best Suchi in town",
        "lat": "7.8833043", (?)
        "lon": "98.3507689", (?)
        "type": "Dinner",
        "expense": "100",
        "rating": 3.7,
        "itinerary_id": 1

      }
  },
  {
      "id": "1",
      "type": "activities"
      "attributes": {
        "name": "Boat Ride"
        "address": "123 Coast Street, Phuket, Thailand",
        "description": "Best Coral Reef boat ride",
        "lat": "7.8833043", 
        "lon": "98.3507689", 
        "type": "Dinner",
        "expense": "100",
        "rating": 4.7,
        "itinerary_id": 1
      }
    }
  ]
}
```
```
status: 404
{
  "errors": [
    {
        "detail": "Couldn't find Itinerary with 'id'=123123123123"
    }
  ]
 }
```

### PR DESCRIPTION
**Pull request by yainporter**
## Type of change
- [x] New feature
- [ ] Bug Fix
- [x] Test
- [ ] Refactor/Style

## Description
Relevant User Story(s):
*Brief summary of the change(s) included in this pull request*
### Activities PATCH, POST, DESTROY and Accommodations DESTROY
- Removed TIME from DailyItineraries, times should be on Activities
- Add TIME to Activities
- Deleted activity type, have activity type and name seemed redundant?
- Only required the name to create an activity
- Fixed spacing on other pages

This description closes #7 

## Check the correct boxes
- [x] This broke nothing
- [ ] This broke some stuff
- [ ] This broke everything

## Testing Changes
- [ ] No Tests have been changed
- [x] Some Tests have been changed
- [ ] All of the Tests have been changed

*Please describe what in the world happened if tests are broken*
For some reason I was getting an error here for Trips, so I changed the expected to match
```
     Failure/Error: expect(create_response[:errors].first[:detail]).to eq("Validation failed: End date must be greater than 2025-12-10 12:00:00 UTC")
     
       expected: "Validation failed: End date must be greater than 2025-12-10 12:00:00 UTC"
            got: "Validation failed: End date must be greater than 2025-12-10"
     
       (compared using ==)
     # ./spec/requests/api/v1/trips_requests_spec.rb:222:in `block (3 levels) in <top (required)>'
```
## Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have fully tested my code

### CODE REVIEW

File: db/migrate/20240410213520_remove_time_from_daily_itinerary.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class RemoveTimeFromDailyItinerary < ActiveRecord::Migration[7.1]
+  def change
+    remove_column :daily_itineraries, :time
+    add_column :activities, :time, :time
```
Comment (IgorrMagalhaess) [inline db/migrate/20240410213520_remove_time_from_daily_itinerary.rb:4]: Awesome!

File: spec/models/activity_spec.rb [hunk 2]
```diff
@@ -5,4 +5,4 @@
       it { should belong_to(:daily_itinerary) }
       it { should have_many(:trip).through(:daily_itinerary) }
    end
-end
```
Comment (IgorrMagalhaess) [inline spec/models/activity_spec.rb:8]: We should add the validations in the tests as well

**Review by IgorrMagalhaess — APPROVED** [review]

File: spec/models/activity_spec.rb (hunk 2 above)
Comment (yainporter) [reply spec/models/activity_spec.rb:8]: Thanks for the reminder!

Comment (yainporter) [reply spec/models/activity_spec.rb:8]: I've re-pushed

**Review by hoadam — APPROVED** [review]

**Pull request merged**